src/Block/BlockContext.php

```diff
@@ -65,6 +65,6 @@ public function setSetting(string $name, $value): BlockContextInterface
 
     public function getTemplate(): string
     {
-        return $this->getSetting('template');
+        return (string) $this->getSetting('template');
```

Comment:
What are you trying to fix ?

A `null` template ?

If the setting has something else than `string|null`, this seems to be an error.

Reply:
Exactly that. If `null` it'll result in a fatal error because the function expects it to always be a `string` while this may strictly not be the case.

Reply:
It seems better to fix the typehint and update to `?string` then

Reply:
Well, I noticed that for example the `\Sonata\BlockBundle\Block\Service\AbstractBlockService::execute` requires this to be a `string` so I did not want to break that. Perhaps we can introduce a `hasTemplate` function so this can be checked properly if need be?

Reply:
You mean
```
$this->renderResponse($blockContext->getTemplate(), [
            'block_context' => $blockContext,
            'block' => $blockContext->getBlock(),
        ], $response);
```
?

Both 
```
$this->renderResponse(null, [])
```
and 
```
$this->renderResponse('', [])
```
Will return an error.

So I would say, let's first fix the `getTemplate` return typehint.
And then if some call are not checking `null`, we'll fix them. But these calls wasn't working before so we're not breaking something.

Reply:
In that case we can just fix the typehint and update it to `?string` as you suggested.

Reply:
@VincentLanglet I just noticed that the allowed types for `template` are mixed (`['string', 'bool']`) so setting the return type to `?string` doesn't always work.

Reply:
Can you provide me some links about this `bool` value ?

Because when I looked at the usage, 
```
$this->renderResponse($blockContext->getTemplate(), [
            'block_context' => $blockContext,
            'block' => $blockContext->getBlock(),
        ], $response);
```
A bool value would lead to an error...

Reply:
https://github.com/sonata-project/SonataBlockBundle/blob/9724f2f658b329a69889fcd61d0ba363926d87e4/src/Block/BlockContextManager.php#L176

Reply:
Maybe we should use `template` => null instead then.

Reply:
It's been there for a while https://github.com/sonata-project/SonataBlockBundle/commit/64b3a61cfd830f4853e369f77b57083c0be20b28 but yeah, setting this to `null` makes more sense just by looking at the interface and the annotations.